prov/verbs/src/verbs_info.c

```diff
@@ -1007,6 +1012,162 @@ static int vrb_ifa_rdma_info(const struct ifaddrs *ifa, char **dev_name,
 	return ret;
 }
 
+int vrb_get_port_space(const struct fi_info *info)
+{
+	if (info != NULL && info->addr_format == FI_SOCKADDR_IB)
+		return RDMA_PS_IB;
+	else
+		return RDMA_PS_TCP;
+}
+
+static int vrb_gid_is_null(const union ibv_gid *gid)
+{
+	return !(gid->raw[8]  | gid->raw[9]  | gid->raw[10] | gid->raw[11] |
+	         gid->raw[12] | gid->raw[13] | gid->raw[14] | gid->raw[15]);
+}
+
+static int vrb_gid_to_str(const union ibv_gid *gid, char gid_str[40])
+{
+	if (sprintf(gid_str, "%02x%02x:%02x%02x:%02x%02x:%02x%02x:%02x%02x:%02x%02x:%02x%02x:%02x%02x",
+		        gid->raw[ 0], gid->raw[ 1],
+		        gid->raw[ 2], gid->raw[ 3],
+		        gid->raw[ 4], gid->raw[ 5],
+		        gid->raw[ 6], gid->raw[ 7],
+		        gid->raw[ 8], gid->raw[ 9],
+		        gid->raw[10], gid->raw[11],
+		        gid->raw[12], gid->raw[13],
+		        gid->raw[14], gid->raw[15]) != 39)
+	{
+		return -FI_EINVAL;
+	}
+	return 0;
+}
+
+static int vrb_gid_rdma_info(char gid_str[40], char **dev_name,
+                             struct rdma_addrinfo **rai)
+{
+	struct rdma_cm_id *id = NULL;
+	struct rdma_addrinfo rai_hints = {
+		.ai_flags = RAI_PASSIVE | RAI_NUMERICHOST | RAI_FAMILY,
+		.ai_family = AF_IB,
+		.ai_port_space = RDMA_PS_IB,
+	}, *rai_ = NULL;
+	int ret;
+
+	ret = rdma_create_id(NULL, &id, NULL, rai_hints.ai_port_space);
+	if (ret) {
+		ret = -errno;
+		goto err;
+	}
+
+	ret = rdma_getaddrinfo(gid_str, NULL, &rai_hints, &rai_);
+	if (ret) {
+		ret = -errno;
+		goto err;
+	}
+
+	ret = rdma_bind_addr(id, rai_->ai_src_addr);
+	if (ret) {
+		ret = -errno;
+		goto err;
+	}
+
+	if (!id->verbs) {
+		ret = -FI_EINVAL;
+		goto err;
+	}
+
+	*dev_name = strdup(ibv_get_device_name(id->verbs->device));
+	if (!(!dev_name)) {
+		ret = -FI_ENOMEM;
+		goto err;
+	}
+
+	rdma_destroy_id(id);
+	*rai = rai_;
+	return 0;
+
+err:
+	if (rai_)
+		rdma_freeaddrinfo(rai_);
+	if (id)
+		rdma_destroy_id(id);
+	return ret;
+}
```

Comment:
Can the above function be replaced with filling out a struct rdma_addrinfo locally and avoid calling into the librdamcm to fill this out?

Reply:
@shefty would you please detail a bit your suggestion? I'm not sure to correcly understand your point.